src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -396,6 +397,50 @@ private void VerifyIsAlive()
                 throw new InvalidOperationException(SR.AssemblyLoadContext_Verify_NotUnloading);
             }
         }
+
+        private static readonly AsyncLocal<AssemblyLoadContext> _asyncLocalCurrentContextualReflectionContext = new AsyncLocal<AssemblyLoadContext>();
+
+        public static AssemblyLoadContext CurrentContextualReflectionContext
+        {
+            get { return _asyncLocalCurrentContextualReflectionContext.Value; }
+        }
+
+        public ContextualReflectionScope EnterContextualReflection()
+        {
+            return new ContextualReflectionScope(this);
+        }
+
+        static public ContextualReflectionScope EnterContextualReflection(Assembly activating)
+        {
+            return activating != null ?
+                GetLoadContext(activating).EnterContextualReflection() :
+                new ContextualReflectionScope(null);
+        }
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public struct ContextualReflectionScope : IDisposable
+        {
+            private bool disposed;
+            private AssemblyLoadContext predecessor;
+
+            public ContextualReflectionScope(AssemblyLoadContext activating)
```

Comment:
Is this constructor meant to be public ? It is duplicate of the `EnterContextualReflection`

Reply:
It doesn't need to be public. I wasn't really thinking about it because I wasn't intending to expose any members in the ref façade.  I will change.